AcademyApp/ViewControllers/LectureDetailViewController.swift

```diff
@@ -0,0 +1,138 @@
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: true)
+    }
+}
+
+// MARK: - Public methods
+
+extension LectureDetailViewController {
+    func configure(with lecture: Lecture) {
+        self.lecture = lecture
+    }
+}
+
+// MARK: - UICollectionViewDataSource methods
+
+extension LectureDetailViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return Sections.allCases.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let section = Sections(rawValue: indexPath.row) else {
+            fatalError("Unable to detec section from initializer ")
+        }
+
+        guard let lecture = lecture else {
+            fatalError("Could not find lecture for detail")
+        }
+
+        switch section {
+
+        case .header:
+            let cell: LectureHeaderCollectionViewCell  = collectionView.dequeReusableCell(for: indexPath)
+            cell.configure(with: lecture)
+            return cell
+        case .buttons:
+            let cell: LectureAttendenceCollectionViewCell  = collectionView.dequeReusableCell(for: indexPath)
+            return cell
+        case .content:
+            let cell: LectureDescriptionCollectionViewCell  = collectionView.dequeReusableCell(for: indexPath)
+            return cell
+        }
+    }
+}
+
+// MARK: - UICollectionView Delegate methods
+
+extension LectureDetailViewController: UICollectionViewDelegate {
+
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        // If scroll view is scrolled to the bottom, hide gradient layer at the bottom of the screen. Otherwise
+        // show it.
+        maskingView.isHidden = collectionView.isScrolledToBottom()
+    }
+}
+
+// MARK: - Private setup methods
+
+private extension LectureDetailViewController {
+
+    func setup() {
+        setupCollectionView()
+        setupGradientLayer()
+        view.backgroundColor = .almostBlack
+    }
+
+    func setupCollectionView() {
+        collectionView.delegate = self
+        collectionView.dataSource = self
+
+        collectionView.register(LectureHeaderCollectionViewCell.self)
+        collectionView.register(LectureAttendenceCollectionViewCell.self)
+        collectionView.register(LectureDescriptionCollectionViewCell.self)
+
+        collectionView.showsVerticalScrollIndicator = false
+        collectionView.backgroundColor = .almostBlack
+
+        // Allow colelction view to go under navigation bar.
+        collectionView.contentInsetAdjustmentBehavior = .never
+
+        if let layout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            layout.estimatedItemSize = CGSize(width: 50, height: 50)
```

Comment:
this estimate is very off - width will be always full width

Reply:
Changed to UICollectionViewFlowLayout.automaticSize.